lib/internal/source_map/source_map_cache.js

```diff
@@ -148,17 +152,39 @@ function rekeySourceMap(cjsModuleInstance, newInstance) {
   }
 }
 
+// WARNING: The `sourceMapCacheToObject` and `appendCJSCache` run during
+// shutdown. In particular, they also run when Workers are terminated, making
+// it important that they do not call out to any user-provided code, including
+// built-in prototypes that might have been tampered with.
+const ObjectCreate = Object.create;
```

Comment:
I personally prefer to destruct this along with the Object because it gives you an idea about what primordials are used in this file from a simple glance, also because I tend to look at the top of the file to understand what kind of world this particular file lives in.

Reply:
I have the same preference. And I wonder if we shouldn't just put everything on `primordials`:
`primordials.ObjectCreate`, `primordials.ObjectHasOwnProperty`, ...

Reply:
I wonder if we could clean up the logic a bit, by simply creating a `internal/source_map/safe_object.js` that encapsulates this logic ... if we find we want to use this in additional places, we could move it to some utility folder in the future.

Reply:
@bcoe already done, these come from the `primordials` global in node which is a safe copy of all the globals and their properties.

Reply:
@joyeecheung I’ve moved this to the top of the file and used destructuring as much as possible, let me know if that’s what you had in mind.